third_party/move/move-model/src/ast.rs

```diff
@@ -1515,6 +1818,54 @@ impl Operation {
         )
     }
 
+    /// Determines whether this op is side-effect free.
+    pub fn is_side_effect_free(&self) -> bool {
+        use Operation::*;
+        matches!(
```

Comment:
A match with exhaustive listing of match arms might be nice here, so that new additions to operations do not go silently missed.